### ISSUE TITLE
Rate Matrix test bug

### ISSUE THREAD

**Issue by kyleabeauchamp** [msmbuilder/msmbuilder#508]
Pulled from Travis on #507

```
======================================================================

ERROR: msmbuilder.tests.test_ratematrix.test_optimize_1

----------------------------------------------------------------------

Traceback (most recent call last):

File "/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest

self.test(*self.arg)

File "/home/travis/miniconda/envs/_test/lib/python2.7/site-packages/msmbuilder-3.2.0.dev0-py2.7-linux-x86_64.egg/msmbuilder/tests/test_ratematrix.py", line 285, in test_optimize_1

y, x, n = model.loglikelihoods_.T
AttributeError: 'list' object has no attribute 'T'
```

Digging into this, I found the following code in ratematrix that doesn't quite make sense to me:

```
        def objective(theta):
            f, g = _ratematrix.loglikelihood(theta, countsmat, lag_time)
            loglikelihoods.append(f)

            if not np.isfinite(f):
                f = np.nan

            loglikelihoods.append(f)
            return -f, -g
```

E.g. I'm confused why f gets added twice.  Probably need a docstring or comment on shapes and dtypes for the list of likelihoods or something.

**Comment by kyleabeauchamp** [comment]
Oops, error message didn't get pasted, fixed now.

```
AttributeError: 'list' object has no attribute 'T'
```

**Comment by rmcgibbo** [comment]
just a mistake. should only be added once to that list.

**Comment by kyleabeauchamp** [comment]
1.  Do you intend to append before or after the inf to nan conversion?
2.  what's going on in the test?

```
    y, x, n = model.loglikelihoods_.T
    x = x-x[0]
    cross = np.min(np.where(n==n[-1])[0])
```

**Comment by kyleabeauchamp** [comment]
OK, I made some guesses about what this is supposed to be and implemented them in #507
